openmc/summary.py

```diff
@@ -567,92 +403,8 @@ def get_material_by_id(self, material_id):
 
         """
 
-        for material in self.materials.values():
+        for material in self.materials:
             if material.id == material_id:
                 return material
 
         return None
-
```

Comment:
I propose that the `get_all_*` method always return a dictionary mapping IDs to things (unlike what `Geometry.get_all_*` currently does). If the user really wants a list of things, they can just do `list(__.get_all_things().values())`. To get a cell by ID would be `geometry.get_all_cells()[my_id]`.

Alternatively, we could always return a list of objects. That way, we could leave it up to the user what kind of dictionary they want, e.g.
```Python
cell_by_name = {c.name: c for c in geometry.get_all_cells()}
```

So, whatever we do, we should make sure that `Geometry.get_all_cells()` and `Universe.get_all_cells()` are consistent since they are not currently. Let me know what you think.